packages/web-pkg/src/composables/piniaStores/modals.ts

```diff
@@ -0,0 +1,90 @@
+import { ContextualHelperData } from 'design-system/src/helpers'
+import { defineStore } from 'pinia'
+import { v4 as uuidV4 } from 'uuid'
+import { Component, ComponentPublicInstance, computed, ref, unref } from 'vue'
+
+export type CustomModalComponent = Component<{ modal: Modal } & Record<string, any>>
+
+export type CustomModalComponentEmits = {
+  confirm?: () => Promise<void>
+  cancel?: () => void
+  'update:confirmDisabled': (value: boolean) => void
+} & Record<string, any>
```

Comment:
Can't we pass this to `defineComponent` when defining our components? Could give us type safety for emitting events etc

Reply:
I tried that without success. As we talked about earlier, the typing of these components can definitely be improved in the future. Maybe we should go on a hunt for good docs about this whole topic 👀